02_bubble/pfannerj/main_test.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	mainout = &buf
+
+	main()
+
+	expected := `Bubble sort starting...
+[1 2 3 5]
+Bubble sort completed...
+Insertion sort starting...
+[1 2 3 5]
+Insertion sort completed...
+`
+	actual := buf.String()
+	if expected != actual {
+		t.Errorf(expected, actual, "Unexpected output in main()")
+	}
+}
+
+func TestBubbleOutputFour(t *testing.T) {
+
+	slice := bubble([]int{3, 2, 1, 5})
+
+	expected := [4]int{1, 2, 3, 5}
+	actual := [4]int{}
+	for i, v := range slice {
+		actual[i] = v
+	}
+	if expected != actual {
+		t.Errorf("Unexpected output in bubble()")
+	}
+}
```

Comment:
Consider using table driven test cases to avoid duplication of test cases. Please see link below for example. 
https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
Yeah, will do... I only saw that in the collected comments after I submitted the PR... that it is definitely a much better way to setup the tests